cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -1117,7 +1192,6 @@ void extend(
  * @param[in] sample_filter an optional device filter function object that greenlights samples
  * for a given query. (none_sample_filter for no filtering)
  */
-
 void search(raft::resources const& res,
```

Comment:
Not necessarily needed for this PR, but it would be nice if we could work towards having a more agnostic API for the `composite::index` that could accept any `composite::index` and be able to search it appropriately (aka dispatch to the appropriate index type). Especially for the next step, when we move towards merging the`tiered::index` together and then searching it.

Reply:
Hi @cjnolet @achirkin , I made an index refactoring. Could you take a moment to review? Many thanks!